src/proto/streams/streams.rs

```diff
@@ -732,7 +732,7 @@ where
         let recv_closed = stream.state.is_recv_closed();
         me.counts.transition(stream, |_, stream|
             // if this is the last reference to the stream, reset the stream.
-            if stream.ref_count == 0 && recv_closed {
+            if stream.ref_count == 0 && recv_closed == false {
```

Comment:
I think a `!recv_closed` would be more idiomatic.

Reply:
Hmm, I generally prefer `== false` in complex conditionals – I feel like it's easy to miss a single exclamation point but much harder to miss the equality check. But I'm happy to change it.